### PR TITLE
Add template args to `DataBox`

### PR DESCRIPTION
**Pull request by brryan**
We started getting compiler errors from `DataBox` due to lack of template argument. I added `<Real>` to the neutrino and photon `DataBox`es used by the mean opacity functions.

### CODE REVIEW

**Comment by brryan** [comment]
I'm not sure what's  going on with the CI here `E: Failed to fetch mirror+file:/etc/apt/apt-mirrors.txt/pool/main/c/curl/libcurl4-openssl-dev_7.81.0-1ubuntu1.14_amd64.deb  404  Not Found [IP: 40.81.13.82 80]` @Yurlungur @mauneyc-LANL

**Comment by Yurlungur** [comment]
> I'm not sure what's going on with the CI here `E: Failed to fetch mirror+file:/etc/apt/apt-mirrors.txt/pool/main/c/curl/libcurl4-openssl-dev_7.81.0-1ubuntu1.14_amd64.deb 404 Not Found [IP: 40.81.13.82 80]` @Yurlungur @mauneyc-LANL

I bet the VM changed. Let me see if I can fix it

**Comment by Yurlungur** [comment]
@brryan I fixed the CI--needed to update the workflows clone call. Some bits have rotted with the tests, though, in particular with ports-of-call.

**Review by Yurlungur — APPROVED** [review]
LGTM

**Comment by brryan** [comment]
Thanks @Yurlungur! Can I go ahead and merge even with the `ports-of-call` issue in CI, or are you still looking at that?

**Comment by Yurlungur** [comment]
I haven't figure it out. Something screwy with the cmake. I may need @mauneyc-LANL 's help

**Comment by cmauney** [comment]
Can you clarify is this `spiner` or `ports-of-call`?

**Comment by cmauney** [comment]
Looking over `cmake/SetupDeps.cmake`, I don't see the `spiner` target added to `singularity-opaq::flags`, and the tests don't bring in `spiner::spiner`, which is probably why the tests are giving the include errors.

**Comment by Yurlungur** [comment]
> Can you clarify is this `spiner` or `ports-of-call`?

There were a number of issues. With an update to `spiner`, `singularity-opac` needed a copy of `ports-of-call` again. But also HDF5 is being a problem. Whether or not spiner should use it is not being appropriately communicated between `singularity-opac` and `spiner`.

**Comment by Yurlungur** [comment]
OK I think I fixed it. @brryan @cmauney @mauneyc-LANL please take a look.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# © 2021. Triad National Security, LLC. All rights reserved.  This
+# © 2023. Triad National Security, LLC. All rights reserved.  This
```
Comment (brryan) [inline CMakeLists.txt]: 2021-2023?

File: cmake/SetupDeps.cmake [hunk 2]
```diff
@@ -64,9 +66,22 @@ if (HDF5_FOUND)
       else()
 	message("MPI::MPI_CXX provided by parent package")
       endif()
+    else()
+      message(status "HDF5 is not parallel")
```
Comment (brryan) [inline cmake/SetupDeps.cmake:70]: `message(STATUS`?

File: singularity-opac/photons/mean_opacity_photons.hpp [hunk 3]
```diff
@@ -186,8 +186,8 @@ class MeanOpacity {
   PORTABLE_INLINE_FUNCTION Real fromLog_(const Real lx) const {
     return std::pow(10., lx);
   }
-  Spiner::DataBox lkappaPlanck_;
-  Spiner::DataBox lkappaRosseland_;
+  Spiner::DataBox<Real> lkappaPlanck_;
```
Comment (brryan) [inline singularity-opac/photons/mean_opacity_photons.hpp]: Thanks for fixing the `mean_s_opacity` functions as well -- should we move these original changes to the `using Databox = ` pattern you have elsewhere for consistency? I can do this if you want to hand the PR back to me

**Review by brryan — COMMENTED** [review]
LGTM

**Comment by cmauney** [comment]
I still haven't wrapped my head around the original issue, but I don't see any problems with changes.If it works g2g

File: cmake/SetupDeps.cmake (hunk 2 above)
Comment (cmauney) [reply cmake/SetupDeps.cmake:70]: case-senstivity is weird in CMake but I think in this case it's insensitive

Comment (brryan) [reply cmake/SetupDeps.cmake:70]: I briefly googled this out of curiosity and the first example was `message(status` of all things and apparently it actually does matter in this case: https://stackoverflow.com/a/35034125

File: CMakeLists.txt (hunk 1 above)
Comment (Yurlungur) [reply CMakeLists.txt]: :+1:

File: cmake/SetupDeps.cmake (hunk 2 above)
Comment (Yurlungur) [reply cmake/SetupDeps.cmake:70]: It looks like that stack overflow is about the case of the thing the message is about, not the STATUS flag? The rest of the code is using lower-case, which is why I kept doing it

File: singularity-opac/photons/mean_opacity_photons.hpp (hunk 3 above)
Comment (Yurlungur) [reply singularity-opac/photons/mean_opacity_photons.hpp]: Sure, that sounds good. Thanks!

**Comment by Yurlungur** [comment]
@brryan if you wanna change the `mean_opacity` to the `using` pattern, go for it. Or just merge at will, either way is good with me. :+1:

**Pull request merged**